backend/data/scrapers/gened_scraper.py

```diff
@@ -0,0 +1,68 @@
+"""
+Program extracts raw data for Programs and place data in file
+'coursesPureRaw.json', ready for formatting.
+
+Step in the data's journey:
+    [ X ] Scrape raw data (gened_scraper.py)
+    [   ] Format scraped data
+    [   ] Customise formatted data
+"""
+
+import json
+
+import requests
+from data.scrapers.courses_scraper import TOTAL_COURSES
+from data.scrapers.payload import HEADERS, create_payload_gened
+from data.utility import data_helpers
+
+URL = "https://www.handbook.unsw.edu.au/api/es/search?appliedFilters="
+
+def scrape_gened_data(year=None):
+    """
+    Retrieves gen ed data for all undergraduate programs
+    Stores in genedPureRaw.json as list of course codes for each program code in programsFormattedRaw.json
+    """
+    faculty_list = []
+    gened_raw = {}
+    cl_id = ""
+    academicOrg = ""
+
+    programs_formatted = data_helpers.read_data("data/scrapers/programsFormattedRaw.json")
+    for program_code in programs_formatted:
+        program = programs_formatted[program_code]
+        faculty = program["faculty"]
+        if faculty in faculty_list:
+            continue
+        else:
+            faculty_list.append(faculty)
```

Comment:
since we don't care for order of faculties and would ideally like to prevent duplicates from accidentally occuring if this ever gets expanded in future, it may be safer to declare `faculty` as a set.
```py
# faculty_list = [] becomes:
faculty_list = set()
# then this if-else can be replaced with
faculty_list.add(faculty)```